sklearn/inspection/tests/test_partial_dependence.py

```diff
@@ -440,6 +455,115 @@ def test_partial_dependence_pipeline():
+         (RobustScaler(), [iris.feature_names[i] for i in (1, 3)]))],
+    ids=['None', 'column-transformer']
+)
+@pytest.mark.parametrize(
+    "features",
+    [[0, 2], [iris.feature_names[i] for i in (0, 2)]],
+    ids=['features-integer', 'features-string']
+)
+def test_partial_dependence_dataframe(estimator, preprocessor, features):
+    # check that the partial dependence support dataframe
+    pd = pytest.importorskip("pandas")
+    df = pd.DataFrame(iris.data, columns=iris.feature_names)
+
+    pipe = make_pipeline(preprocessor, estimator)
+    pipe.fit(df, iris.target)
+    pdp_pipe, values_pipe = partial_dependence(
+        pipe, df, features=features, grid_resolution=10
+    )
+
+    # the column transformer will reorder the column when transforming
+    # we mixed the index to be sure that we are computing the partial
+    # dependence of the right columns
+    if preprocessor is not None:
+        X_proc = clone(preprocessor).fit_transform(df)
+        features_clf = [0, 1]
+    else:
+        X_proc = df
+        features_clf = [0, 2]
+
+    clf = clone(estimator).fit(X_proc, iris.target)
+    pdp_clf, values_clf = partial_dependence(
+        clf, X_proc, features=features_clf, method='brute', grid_resolution=10
+    )
+
+    assert_allclose(pdp_pipe, pdp_clf)
+    if preprocessor is not None:
+        scaler = preprocessor.named_transformers_['standardscaler']
+        assert_allclose(
+            values_pipe[1],
+            values_clf[1] * scaler.scale_[1] + scaler.mean_[1]
+        )
+    else:
+        assert_allclose(values_pipe[1], values_clf[1])
+
+
+@pytest.mark.parametrize(
+    "features, expected_pd_shape",
+    [(0, (3, 10)),
+     (iris.feature_names[0], (3, 10)),
+     ([0, 2], (3, 10, 10)),
+     ([iris.feature_names[i] for i in (0, 2)], (3, 10, 10)),
+     (slice(0, 2, 1), (3, 10, 10)),
+     ([True, False, True, False], (3, 10, 10))],
+    ids=['scalar-int', 'scalar-str', 'list-int', 'list-str', 'slice', 'mask']
+)
+def test_partial_dependence_feature_type(features, expected_pd_shape):
+    # check all possible features type supported in PDP
+    pd = pytest.importorskip("pandas")
+    df = pd.DataFrame(iris.data, columns=iris.feature_names)
+
+    preprocessor = make_column_transformer(
+        (StandardScaler(), [iris.feature_names[i] for i in (0, 2)]),
+        (RobustScaler(), [iris.feature_names[i] for i in (1, 3)])
+    )
+    pipe = make_pipeline(
+        preprocessor, LogisticRegression(max_iter=1000, random_state=0)
+    )
+    pipe.fit(df, iris.target)
+    pdp_pipe, values_pipe = partial_dependence(
+        pipe, df, features=features, grid_resolution=10
+    )
+    assert pdp_pipe.shape == expected_pd_shape
+    assert len(values_pipe) == len(pdp_pipe.shape) - 1
+
+
+@pytest.mark.parametrize(
+    "name, Estimator", all_estimators(type_filter=['classifier', 'regressor'])
+)
+def test_partial_dependence_unfitted(name, Estimator):
```

Comment:
How does this relate to dataframes support?

Reply:
I change the code and make a consistent `NotFittedError` which should fail early in all cases (tree based and brute method). I wanted to be sure that the error is properly raised for all estimators.

Here, I just want to make sure that the error is consistent for:

- a single estimator
- a pipeline (this does not support check_is_fitted)
- and the case that we have a ColumnTransformer because it is a more complex nested business

Reply:
Can we have a smaller list of estimators to run this test? (Instead of `all_estimators`?)

Reply:
It is quite fast as a test, less than 1 sec for executing everything (with initial pytest collection).
The idea is to check the error message consistency across all classifiers and regressors. Not using all of them open the doors for inconsistencies. What would you have in mind as a sub-lists?

Reply:
I think I agree with Thomas. The current test is more of an exhaustive test for `check_is_fitted` than a test for the PDP code. I think a test with a single estimator and a pipeline should be enough.

Typically in meta-estimators tests, we don't use `all_estimators`.

Reply:
fair enough